tests/init_arg/canbench.yml

```
@@ -0,0 +1,8 @@
+build_cmd:
+  cargo build --release --target wasm32-unknown-unknown
+
+init:
+  4449444c0001710568656c6c6f
```

How about we do this:

1. Rename `init` to `init_args` for clarity.
2. Ideally, we'd want to support candid records as well, so specifying a type will allow to add that support in a backward-compatibly way afterwards.

```suggestion
init_args:
  hex:
    4449444c0001710568656c6c6f
```

---

@ielashi: I've change the config as requested. In order to support nested yaml maps I changed the whole config parsing to typed structs rather than just parsing into a `BTreeMap<String, String>`.
PTAL